### PR TITLE
Extend GraphQL customer information with login status.

### PR DESCRIPTION
**Pull request by lifedraft**
Unfortunately by calling the Commerce_Customer query this failed for not authenticated users. From Frontend perspective this query should be more resilient, should not fail and return users login state. Therefore this MR extends the GraphQL response of this query.

TBD; We could remove Commerce_Customer_Status query in future major release as this becomes obsolete?

### CODE REVIEW

File: customer/interfaces/graphql/schema.graphql [hunk 1]
```diff
@@ -4,7 +4,9 @@ type Commerce_Customer_Status_Result {
 }
 
 type Commerce_Customer_Result {
-    id: String!
+    id: String! @deprecated(reason: "Use `userID` field.")
+    userID: String!
+    isLoggedIn: Boolean!
```
Comment (bastianccm) [inline customer/interfaces/graphql/schema.graphql:9]: isLoggedIn is semantically not making sense here

File: customer/interfaces/graphql/schema.graphql [hunk 2]
```diff
@@ -4,7 +4,9 @@ type Commerce_Customer_Status_Result {
 }
 
 type Commerce_Customer_Result {
-    id: String!
+    id: String! @deprecated(reason: "Use `userID` field.")
+    userID: String!
```
Comment (bastianccm) [inline customer/interfaces/graphql/schema.graphql:8]: What is the difference between userID and id here? Why is it deprecated? "Use another field" is not a reason

File: customer/interfaces/graphql/queryresolver.go [hunk 3]
```diff
@@ -46,12 +47,14 @@ func (r *CustomerResolver) CommerceCustomerStatus(ctx context.Context) (*dtocust
 // CommerceCustomer resolver the commerce customer
 func (r *CustomerResolver) CommerceCustomer(ctx context.Context) (*dtocustomer.CustomerResult, error) {
 	user, err := r.service.GetForIdentity(ctx, web.RequestFromContext(ctx))
-	if err != nil {
-		return nil, err
+	if errors.Is(err, application.ErrNoIdentity) {
+		return &dtocustomer.CustomerResult{IsLoggedIn: false}, nil
```
Comment (bastianccm) [inline customer/interfaces/graphql/queryresolver.go:51]: Simply do `return nil, nil`, to indicate that there is no Customer, but also no error.

File: customer/interfaces/graphql/dtocustomer/dto.go [hunk 4]
```diff
@@ -14,6 +14,8 @@ type (
 	// CustomerResult is a dto
 	CustomerResult struct {
 		ID                     string
+		UserID                 string
+		IsLoggedIn             bool
```
Comment (bastianccm) [inline customer/interfaces/graphql/dtocustomer/dto.go:18]: See the GraphQL Schema for comments

File: customer/interfaces/graphql/schema.graphql (hunk 2 above)
Comment (lifedraft) [reply customer/interfaces/graphql/schema.graphql]: removed.

File: customer/interfaces/graphql/queryresolver.go (hunk 3 above)
Comment (lifedraft) [reply customer/interfaces/graphql/queryresolver.go]: changed.

File: customer/interfaces/graphql/dtocustomer/dto.go (hunk 4 above)
Comment (lifedraft) [reply customer/interfaces/graphql/dtocustomer/dto.go]: also removed.

**Pull request merged**